upgrade/UPGRADE-v8.1.0-dev.md

```diff
@@ -266,6 +266,63 @@ There you can find links to upgrade notes for other versions too.
     - some of the issues related to class extension need to be addressed manually nevertheless (see the ["Framework extensibility" article](../introduction/framework-extensibility.md#problem-3)) for more information
     - you need to resolve all the other reported problems (it is up to you whether you decide to address them directly or add ignores in your `phpstan.neon`). You can find inspiration in [#1381](https://github.com/shopsys/shopsys/pull/1381) and [#1040](https://github.com/shopsys/shopsys/pull/1040)
+
+- if you want to add stylelint rules to check style coding standards [#1511](https://github.com/shopsys/shopsys/pull/1511)
+    -  add new file [.stylelintignore](https://github.com/shopsys/shopsys/blob/master/project-base/.stylelintignore)
+    -  add new file [.stylelintrc](https://github.com/shopsys/shopsys/blob/master/project-base/.stylelintrc)
+    - update `gruntfile.js.twig` and add new task
+        ```diff
+        + stylelint: {
+        +     all: [
+        +         'src/**/*.less',
+        +     ]
+        + },
+
+          watch: {
+              admin: {
+        ```
+        ```diff
+          {% else -%}
+        -     ['frontendLess{{ domain.id }}']
+        +     ['frontendLess{{ domain.id }}','stylelint']
+          {% endif -%}
+        ```
+        ```diff
+          grunt.loadNpmTasks('grunt-spritesmith');
+        + grunt.loadNpmTasks('grunt-stylelint');
+
+        - grunt.registerTask('default', ["sprite:admin", "sprite:frontend", "webfont", "less", "postcss", "legacssy"]);
+        + grunt.registerTask('default', ["sprite:admin", "sprite:frontend", "webfont", "less", "postcss", "legacssy", "stylelint"]);
+        ```
+    - update `package.json`
+        ```diff
+              "grunt-spritesmith": "^6.6.2",
+        +     "grunt-stylelint": "^0.12.0",
+              "grunt-webfont": "^1.7.2",
+              "jit-grunt": "^0.10.0",
+        +     "stylelint": "^11.1.1",
+              "time-grunt": "^1.4.0"
+        ```
+
+    - don't forget to rebuild your grunt file by command `php phing gruntifle` and update your npm dependencies by command `npm install`
+
+    - update `build.xml`
+    ```diff
+    +    <target name="stylelint" description="Run CSS linter.">
+    +        <exec executable="${path.stylelint.executable}" logoutput="true" passthru="true" checkreturn="true">
+    +            <arg path="src/**/*.less"/>
+    +        </exec>
+    +    </target>
+  
+    +    <target name="stylelint-fix" description="Run CSS linter.">
+    +        <exec executable="${path.stylelint.executable}" logoutput="true" passthru="true" checkreturn="true">
+    +            <arg path="src/**/*.less"/>
+    +            <arg value="--fix"/>
+    +        </exec>
+    +    </target>  
+    ```
+  
+  - to fix all your less files in command line by command `npx stylelint src/**/*.less --fix` or `php phing stylelint-fix`
```

Comment:
You can use phing target better

Reply:
Hmm, ok, I removed part with npx, but I think that we shloud start using npm scripts in future. This is not place for  discussion about it, but I would like see separate use case for fornted and backand targets. We shloud use phing for backaned and npm for frontend in my opinions.

Reply:
Hmm, it's interesting point of view to me :)
I see Phing as a tool to lower the amount of needed knowledge with predefined tasks. 

Because it's easier for me to take a look at the list of phing targets and see and run `stylelint-fix` than to have to remember what files should be checked and how the script is named or find it in the documentation.

Because for almost every backend task I can use `./bin/console`, so it seems to be very similar.

But I agree this is not the right place for this discussion :)

Reply:
I understand, but I hope that we will have separately frontend in future and for separately frontend we will not need php, we will not need docker, twig and more things.  And then we can use `npm run` instead `php phing` 

But yes, it is early for this :smile: